proposals/lip-0057.md

```diff
@@ -879,21 +881,22 @@ The definition and rationale for the `isCertificateGenerated` function is part o
 
 ```python
 def isUnlockable(unlockObject: UnlockObject, senderAddress: Address, height: uint32) -> bool:
-    # First consider the case that validator is not punished.
     validatorAddress = unlockObject.validatorAddress
-    lockingPeriod = LOCKING_PERIOD_SELF_STAKING if validatorAddress == senderAddress else LOCKING_PERIOD_STAKING
-
-    # If validator is not punished, normal locking period for staked/self-staked tokens applies.
-    if not isPunished(validatorAddress, height):       
-        if height - unlockObject.unstakeHeight < lockingPeriod:
-            return False
+    if validatorAddress == senderAddress:
+        lockingPeriod = LOCKING_PERIOD_SELF_STAKING
+        punishmentWindow = PUNISHMENT_WINDOW_SELF_STAKING
+    else:
+        lockingPeriod = LOCKING_PERIOD_STAKING
+        punishmentWindow = PUNISHMENT_WINDOW_STAKING
+        
+    if height < unlockObject.unstakeHeight + lockingPeriod:
+        return False
 
-    else: # Validator is punished.
+    if len(validatorStore(validatorAddress).reportMisbehaviorHeights) > 0: # Validator might be punished.
         let lastReportMisbehaviorHeight be the last element of validatorStore(validatorAddress).reportMisbehaviorHeights
         # lastReportMisbehaviorHeight is also the largest element of the reportMisbehaviorHeights array.
-        punishmentWindow = PUNISHMENT_WINDOW_SELF_STAKING if validatorAddress == senderAddress else PUNISHMENT_WINDOW_STAKING
-
-        if height – lastReportMisbehaviorHeight < punishmentWindow and lastReportMisbehaviorHeight < unlockObject.unstakeHeight + lockingPeriod:
+ 
+        if height < lastReportMisbehaviorHeight + punishmentWindow and lastReportMisbehaviorHeight < unlockObject.unstakeHeight + lockingPeriod:
```

Comment:
Basically this is not `<=` because locking and punishment periods have to be intended as 'after how many blocks is the period over', right?

Reply:
also, not elated to this PR, but I think the description of `PUNISHMENT_WINDOW_STAKING` and `PUNISHMENT_WINDOW_SELF_STAKING` in the table could be improved. Instead of 'The punishment time' I would maybe say 'punishment period' or even 'number of blocks' to make it clear that this is a number of blocks, not a time.